### PR TITLE
Prime ideal topology (S48) revising description

### PR DESCRIPTION
**Pull request by GeoffreySangston**
There seem to be some issues with the third paragraph/sentence in https://topology.pi-base.org/spaces/S000048 (the one beginning "Any open $U \subset X$ is open...").

### CODE REVIEW

**Comment by JoeF131** [comment]
That's interesting. The first line claims that $\\{ V_{(x)} : x \in \mathbb{Z}^+\\}$ is a basis of $X$ eventhough it's already the entirety of the topology.  Normally one defines the Zariski topology as the topology generated by the basis $\\{V_I : I \text{ prime ideal }\\}$ and I think the second section tries to explain that which is in that case redunant since the first line uses a different definition for the Zariski topology.

Additionally I found that pi-base doesn't know that [S48](https://topology.pi-base.org/spaces/S000048) is [Spectral](https://topology.pi-base.org/spaces/S75), which follows imediately by the equivalent definition of spectral.

**Comment by GeoffreySangston** [comment]
So maybe instead of 'has a generic point', this trait assert quasi-sober. S48 is also currently missing Alexandrov.

Your comment made me realize pi-base doesn't answer the following queries (though there are still a lot of spaces in pi-base not known to be Quasi-sober or ~Quasi-sober), which could be good to address in a future PR.

[π-Base, Search for `has a generic point +  ~Quasi-sober`](https://topology.pi-base.org/spaces?q=has+a+generic+point+%2B++%7EQuasi-sober)

[π-Base, Search for `has a generic point + hereditarily connected + ~quasi-sober`](https://topology.pi-base.org/spaces?q=has+a+generic+point+%2B+hereditarily+connected+%2B+%7Equasi-sober)

**Comment by prabau** [comment]
@JoeF131 I am not familiar enough with the algebraic side of things, but note that pi-base's notion of Spectral space (P75).
But how do we already know that S48 is spectral?

All pi-base knows about is the definition of spectral as a compact sober space where the compact open sets satisfy some specific properties.  The second paragraph in https://topology.pi-base.org/properties/P000075 talking about Stone duality is just background information, but can't really be relied on for justification. I think.

**Comment by JoeF131** [comment]
I mean [P75](https://topology.pi-base.org/properties/P000075) states "Additionally, spectra of commutative rings endowed with the Zariski topology are spectral spaces and vice versa, although this association is not one-to-one." in the second paragraph.

**Comment by prabau** [comment]
Right, these are difficult results.  For now, we may be better off using the topological characterization.

**Comment by GeoffreySangston** [comment]
What do we think about mentioning somewhere on the page that S48 is the underlying topological space of $\mathrm{Spec}(\mathbb{Z})$ (which I just checked)?

**Comment by prabau** [comment]
> What do we think about mentioning somewhere on the page that S48 is the underlying topological space of Spec ( Z ) (which I just checked)?

Yes.

**Comment by JoeF131** [comment]
I just checked the book {{doi:10.1017/9781316543870}} cited in [P75](https://topology.pi-base.org/properties/P000075). The proof that Spec(A) is a spectral space is given in section 2.5. and uses purely topology (with the exception of the definition of the Zariski topology). 
The other direction that for every spectral space there is a ring A such that spec(A) is homeomorphic to the space is given in section 12.6. They do however follow Hochster's result, which is the paper I linked above.

The description of [P75](https://topology.pi-base.org/properties/P000075) exactly states the statements above while refering to {{doi:10.1017/9781316543870}}, so it follows from the current description that S48 is spectral. As such quasi-sober would also follow.

**Comment by GeoffreySangston** [comment]
I have an alternative suggestion (for a future PR). S48 has an algebraic description because it appears in Steen-Seebach that way, and I think that's why it seems natural to assert 'Spectral'. I think can also be described as a kind of 'cone' over the cofinite topology on $\omega$ [(S15)](https://topology.pi-base.org/spaces/S000015) which appends a generic point. I'm not sure what the standard name for this cone construction is, but after briefly looking at [this page](https://en.wikipedia.org/wiki/Extension_topology), I think it's the *closed extension* of S15 by a point. Maybe in order to promote the general topology style of the website, it would be better to define S48 by this, and then refer to the algebraic descriptions via an alias / secondary description?

**Comment by prabau** [comment]
I think both of your suggestions above would be valuable improvements.

Namely, add an algebraic description of the space as $\mathrm{Spec}(\mathbb{Z})$, backed up with a good reference (wikipedia?) showing it's just an example of much more general contruction.  Adding $\mathrm{Spec}(\mathbb{Z})$ as an alias would make sense to me.

And also add a more topological direct description, which is kind of mentioned in item 1 of the counterexample in S&S.

I occasionally see the words "non-Hausdorff cone".  No idea what that is, but would that be related to this construction?

**Comment by GeoffreySangston** [comment]
> Namely, add an algebraic description of the space as Spec ( Z ) , backed up with a good reference (wikipedia?) showing it's just an example of much more general contruction. Adding Spec ( Z ) as an alias would make sense to me.

> And also add a more topological direct description, which is kind of mentioned in item 1 of the counterexample in S&S.

I tried to do this. I may have overstepped by changing the name of the space to prioritize the non-algebraic description. I can easily revert it back and swap the orders of the descriptions if that's what we want. I thought it would be wiser to upload this commit to see what we thought first, and then update the traits to reflect the new primary definition; to that end, hopefully Item #21 of the closed extension topology section will be helpful. 

I removed the reference to Wikipedia for the closed extension topology, because the version from Steen-Seebach is less general than what's on Wikipedia (from what I saw, Steen-Seebach just defines these extensions by adding a single point).

> I occasionally see the words "non-Hausdorff cone". No idea what that is, but would that be related to this construction?

Unfortunately for S48's description / name, the non-Hausdorff cone from the finite topology sources is the open extension topology from Steen-Seebach. The "non-Hausdorff cone point" is a focal point, not a generic point.

File: spaces/S000048/README.md [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
+  - Prime ideal topology
+  - $\mathrm{Spec}(\mathbb{Z})$
 refs:
   - doi: 10.1007/978-1-4612-6290-9 
     name: Counterexamples in Topology
+  - wikipedia: Zariski_topology
+    name: Zariski topology
 ---
-Let $X$ be the set of all prime ideals in $\mathbb{Z}$ with basis consisting of all $V_x = \{P \in X : x \notin P\}$ for all $x \in \mathbb{Z}^+$.
+$X = \omega \sqcup \{p\}$. A nonempty set is open if and only if it contains $p$ and has a finite complement. This is the closed extension of {S15}. See item #20 for space #12 of {{doi:10.1007/978-1-4612-6290-9}}.
```
Comment (prabau) [inline spaces/S000048/README.md:14]:
```suggestion
Let $X = \omega \sqcup \{p\}$. A nonempty set is open if and only if it contains $p$ and has a finite complement. This is the *closed extension* of {S15}.  The closed extension construction is described in general in item #20 for space #12 of {{doi:10.1007/978-1-4612-6290-9}}.
```

File: spaces/S000048/README.md [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
 
-Note that for every $x$, $V_x$ contains all but finitely many $P \in X$ (since each $x$ has finitely many prime factors) and that $0 \in P$ for every $P \in V_x$.
+$X$ is homeomorphic to the set of all prime ideals in $\mathbb{Z}$ with basis consisting of the sets $V_x = \{P \in X : x \notin P\}$ for all $x \in \mathbb{Z}^+$. The zero ideal identifies with the generic point $p \in X$. This is the Zariski topology on $\mathrm{Spec}(\mathbb{Z})$, the prime spectrum of the ring of integers. See {{wikipedia:Zariski_topology}} for the general construction of the Zariski topology on the prime spectrum of a commutative ring.
```
Comment (prabau) [inline spaces/S000048/README.md:16]:
```suggestion
$X$ is homeomorphic to the set of prime ideals of the ring $\mathbb{Z}$ with basis consisting of the sets $V_x = \{P \in X : x \notin P\}$ for all $x \in \mathbb{Z}_{n\ge 0}$. The zero ideal identifies with the generic point $p \in X$. This is the Zariski topology on $\mathrm{Spec}(\mathbb{Z})$, the prime spectrum of the ring of integers. See {{wikipedia:Zariski_topology}} for the general construction of the Zariski topology on the prime spectrum of a commutative ring.
```

**Comment by prabau** [comment]
I think it's fine to have the primary name in non-algebraic terms in this case, and we also have $\mathrm{Spec}(\mathbb Z)$ for the algebraic description.  Both of these precisely identify the space.  The other alias (Prime ideal topology) is the least helpful, but is there because of S&S.  I think it should move to the last position.

Also when referring to a wikipedia entry, the name in the `refs:` section is usually of the form "... on Wikipedia".

**Comment by prabau** [comment]
Should the P201 trait be described in terms of the extra point, now that we emphasize the more topological description?
("Every nonempty open set contains the point $p$.")

**Comment by GeoffreySangston** [comment]
> The name "Closed extension topology" is not used much in the literature, but it does appear in a few places, mainly due to references to S&S. Not sure there is a better way to express things.

Yea this is the issue I found. The most famous representative for this homeomorphism class of spaces is clearly (the underlying space of) $\mathrm{Spec}(\mathbb{Z})$. But it seems like the best description of the space from the point of view of general topology is the direct unwrapping of the definition of the closed extension topology applied to [S15](https://topology.pi-base.org/spaces/S000015). I would guess this name (the closed extension one) is somewhat opaque to most passersby, which is unfortunate. It does have the strength of conveying to the reader that it has something to do with another space that they probably are familiar with. The problem with ' $\mathrm{Spec}(\mathbb{Z})$ ' (or 'Prime Ideal Topology') as the primary name is it sends the reader off into another subject just for one example. (It's not clear if these latter two are less opaque than the closed extension one, except for algebraic geometers.  Still, ' $\mathrm{Spec}(\mathbb{Z})$ ' likely the most recognizable.)

> Should the P201 trait be described in terms of the extra point, now that we emphasize the more topological description? ("Every nonempty open set contains the point p .")

I was thinking about updating all of the traits (or as many as is convenient) to reflect the new definition. Would it be good to do that here? I'll go ahead and change P201 now.

**Comment by prabau** [comment]
Yeah, although not ideal, "Close extension of ..." is probably as good as any.  Unless someone comes up with a better suggestion, let's go with that one.

Up to you if you want to update more traits now, or leave it as is.

**Comment by GeoffreySangston** [comment]
> Up to you if you want to update more traits now, or leave it for a later time.

I think it's good to open this up to other contributors, since I think a few other people wanted to make changes to S48 since this PR appeared.

(I do want to clarify, and I think we agree, that one of the justifications for this change is that it should be easier to directly assert the conditions going into 'Spectral' now, and that I think 'Spectral' should not be asserted as a trait.)

**Comment by david20000813** [comment]
Out of curiosity, since S48 still has spectral being unknown, what is the plan for it now? None of the theorems that implies spectral applies to S48. But S48 is sort of the archetypical example of a spectral space. It feels a bit strange to have that trait being unknown.

**Comment by GeoffreySangston** [comment]
@david20000813 @prabau I think we should add the trait for Quasi-Sober, from which Sober will be deduced by Quasi-Sober + T0. Then we add the trait for [Spectral](https://topology.pi-base.org/properties/P000075) using pi-base's primary definition: 
> A [Compact](https://topology.pi-base.org/properties/P16) [Sober](https://topology.pi-base.org/properties/P73) space for which the collection of compact open subsets is closed under finite intersections and forms a base for the topology.

So the trait-file will mention {S48|P73} and {S48|P16}, and then it will briefly justify all non-empty open sets are compact, so that [Spectral](https://topology.pi-base.org/properties/P000075) follows.

(I'll be honest and say that I did misunderstand [T554](https://topology.pi-base.org/theorems/T000554). I misread 'hereditarily connected' there for 'hyperconnected'. Either way, I think the idea is to use the primary definition of [Spectral](https://topology.pi-base.org/properties/P000075), so that not too many of the basic topological properties are hidden beneath a non-trivial theorem.)

**Comment by david20000813** [comment]
@GeoffreySangston I do wonder if the conditions for T554 can be weakened. I can't even find a counterexample to ["Hyperconnected + Compact + Sober => Spectral"](https://topology.pi-base.org/spaces?q=Hyperconnected+%2B+Compact+%2B+Sober+%2B+not+Spectral). But yes, adding Noetherian does sound like a good idea. Sober + Noetherian does imply Spectral.

(Quasi-sober + $T_0$ is equivalent to Sober, so your last statement is the same as saying Sober + Noetherian => Spectral.)

**Comment by david20000813** [comment]
Also, any (nonempty) spectral space has a closed point, since any nontrivial ring has a maximal ideal which corresponds to closed points under the Zariski topology. Does anyone know a purely topological proof of this fact? This is currently unknown to pi-Base.

**Comment by david20000813** [comment]
Oh, yeah, this follows from Compact + $T_0$ => Has a closed point, which is a generalization of T641. I'll write a PR.

**Comment by david20000813** [comment]
The Alexandrov condition of T554 is necessary. Hereditarily connected also cannot be weakened to hyperconnected.

Hereditarily connected + Compact + Sober does not imply Spectral: Take the right ray topology on $X = [0,1]$, i.e., the open sets are $X$ and $(a,1]$ for $0\leq a\leq 1$. $X$ is clearly hereditarily connected. It is compact because $0$ is a focal point. It is clearly $T_0$. Nonempty closed sets are of the form $[0,a]$ for $0\leq a\leq 1$, which has generic point $a$, so $X$ is sober. It is, however, not spectral, as $(a,1]$ is never compact unless $a=1$ (i.e., when $(a,1]=\varnothing$), so the only compact open sets are $\varnothing$ and $X$, which clearly do not form a basis.

Hyperconnected + Compact + Sober + Alexandrov does not imply Spectral: Consider the set $\\{\frac{1}{n}\\}_{n=1}^\infty$ with its usual order. Let,

$$X=\\{0_a,0_b\\}\cup\\{\frac{1}{n}\\}_{n=1}^\infty$$

where either of the zeros is smaller than all $\frac{1}{n}$ and the two zeros are incomparable. Take the Alexandrov topology on $X$. $X$ is hyperconnected since the ordering on $X$ is upward directed. It is compact since the minimum neighborhood of $0_a$ is $\\{0_a\\} \cup \\{\frac{1}{n}\\}_{n=1}^\infty$ and similarly for $0_b$, and the two neighborhoods already cover the space. $X$ is clearly $T_0$. Closed sets are downward closed sets, so nonempty closed sets are exactly,

$$\\{0_a\\}; \\{0_b\\}; \\{0_a,0_b\\}; \\{0_a,0_b\\}\cup\\{\frac{1}{n}\\}_{n=N}^\infty, \text{ for }N\geq 1$$

$\\{0_a,0_b\\}$ is not irreducible as $\\{0_a\\}$ and $\\{0_b\\}$ are both open in $\\{0_a,0_b\\}$. All other nonempty closed sets are irreducible, and they all have maximum elements, which are generic points. Thus, $X$ is sober.

However, $X$ is not spectral, because,

$$\\{0_a\\} \cup \\{\frac{1}{n}\\}_{n=1}^\infty$$

and,

$$\\{0_b\\} \cup \\{\frac{1}{n}\\}_{n=1}^\infty$$

are both compact open (they both have minimum elements and thus focal points). But their intersection, $\\{\frac{1}{n}\\}_{n=1}^\infty$, is not compact.

**Comment by david20000813** [comment]
While constructing these counterexamples, I realized an interesting fact: The counterexample for Hereditarily connected + Compact + Sober => Spectral has cardinality continuum, and this is necessary - Hereditarily connected + Compact + Sober + Cardinality $<\mathfrak{c}$ => Spectral is actually true. I'll write this in an MSE answer tomorrow, but the basic idea is that the specialization preorder is a bounded linear order with the least upper bound property, so cardinality being strictly smaller than continuum means it cannot be a dense order. Even stronger, no nondegenerate interval can be a dense order. This ensures enough elements have immediate predecessor, which translates to enough compact open sets.

**Comment by GeoffreySangston** [comment]
@david20000813 I loosely read over your write-up for ~20 minutes and it looks really nice. I need to properly read it tomorrow though.

I was wondering if you know any good examples of the following?
[π-Base, Search for `Hereditarily connected + compact + sober + Cardinality $\lt\mathfrak c$ + ~noetherian`](https://topology.pi-base.org/spaces?q=Hereditarily+connected+%2B+compact+%2B+sober+%2B+Cardinality+%24%5Clt%5Cmathfrak+c%24+%2B+%7Enoetherian)

**Comment by david20000813** [comment]
@GeoffreySangston The left ray topology on $\omega + 1$ is an example. It is also Alexandrov. (If you want a non-Alexandrov example, the right open ray topology on $\omega + 1 + \omega^\ast$, where $\omega^\ast$ means $\omega$ with reverse ordering, works.)

**Pull request merged**